ludwig/trainers/trainer_lightgbm.py

```diff
@@ -1057,3 +1074,107 @@ def _construct_lgb_datasets(
             eval_names.append(LightGBMTrainer.TEST_KEY)
 
         return lgb_train, eval_sets, eval_names
+
+
+# Add a top-level function wrapper to handle non-distributed test import errors
+def lightgbm_ray_train_step(
+    model: GBM,
+    params: Dict[str, Any],
+    lgb_train: "RayDMatrix",  # noqa: F821
+    eval_sets: List["RayDMatrix"],  # noqa: F821
+    eval_names: List[str],
+    init_model: lgb.LGBMModel,
+    boost_rounds_per_train_step: int,
+    evals_result: Dict,
+    ray_params: "RayParams",  # noqa
+    evaluate_training_set: bool,
+    device: Optional[str] = None,
+) -> lgb.LGBMModel:
+
+    # We need to add max_calls here to ensure that the Ray actors that get created by the LightGBM class
+    # for each boosting round don't leave dangling resources in the object store memory.
+    @ray.remote(max_calls=1)
```

Comment:
Once the memory leak from `RayLGBMClassifier`/`RayLGBMRegressor` has been fixed, does this mean we can simply remove `max_calls=1`? Would be ideal to add a link to an open issue on the Ray project if one exists.

Reply:
Yup! Ideally, we would remove the ray.remote decorator and move away from the need to use a remote function entirely!